src/click/types.py

```diff
@@ -335,6 +336,23 @@ def shell_complete(
         return [CompletionItem(c) for c in matched]
 
 
+class EnumChoice(Choice):
+    def __init__(self, enum_type: t.Type[enum.Enum], case_sensitive: bool = True):
+        super().__init__(
+            choices=[element.name for element in enum_type],
```

Comment:
- Should there be the ability to use the enum values as choices, enabled through a parameter?
- Should there be the ability to use the enum member itself as the default, which I believe would require adding the enum members themselves to `choices`, not just their string name? `e.g. default=SomeEnum.member`

Reply:
As it turns out [Click officially does not support Enums](https://click.palletsprojects.com/en/8.1.x/parameters/):
> You should only pass a list or tuple of choices. Other iterables (like generators) may lead to surprising results.

I'll revert this back.